tests/unit-tests/dds/DCPS/CompleteToMinimalTypeObject.cpp

```diff
@@ -0,0 +1,607 @@
+                    ::DDS::TopicListener::_nil(),
+                    ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+  ::DDS::Publisher_var pub =
+    dp->create_publisher(PUBLISHER_QOS_DEFAULT,
+                          ::DDS::PublisherListener::_nil(),
+                          ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+  ::DDS::DataWriter_var dw =
+    pub->create_datawriter(my_topic.in (),
+                            DATAWRITER_QOS_DEFAULT,
+                            ::DDS::DataWriterListener::_nil(),
+                            ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+  TypeMap ctm = un_tsi.getCompleteTypeMap();
+  TypeObject cto = ctm[TypeIdentifier(EK_COMPLETE, EquivalenceHashWrapper(80, 3, 53, 153, 247, 134, 18, 76, 42, 156, 63, 245, 137, 180))];
+  TypeMap mtm = un_tsi.getMinimalTypeMap();
+  TypeObject c_to_m_mto;
+  if (!tls->complete_to_minimal_typeobject(cto, c_to_m_mto)) {
+    ASSERT_EQ(1,0);
+  }
+  TypeObject mto2 = mtm[TypeIdentifier(EK_MINIMAL, EquivalenceHashWrapper(115, 7, 58, 161, 22, 145, 40, 98, 164, 53, 75, 57, 105, 170))];
+  ASSERT_EQ(mto2, c_to_m_mto);
+}
+TEST(CompleteToMinimalTypeObject, Enumerated)
+{
+  my_mod::unTypeSupportImpl un_tsi;
+  un_tsi.register_type(dp.in (), "EnumType");
+  ::DDS::Topic_var my_topic =
+    dp->create_topic ("EnumType",
+                    "EnumType",
+                    TOPIC_QOS_DEFAULT,
+                    ::DDS::TopicListener::_nil(),
+                    ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+  ::DDS::Publisher_var pub =
+    dp->create_publisher(PUBLISHER_QOS_DEFAULT,
+                          ::DDS::PublisherListener::_nil(),
+                          ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+  ::DDS::DataWriter_var dw =
+    pub->create_datawriter(my_topic.in (),
+                            DATAWRITER_QOS_DEFAULT,
+                            ::DDS::DataWriterListener::_nil(),
+                            ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+  TypeMap ctm = un_tsi.getCompleteTypeMap();
+  TypeObject cto = ctm[TypeIdentifier(EK_COMPLETE, EquivalenceHashWrapper(208, 102, 40, 140, 43, 232, 192, 255, 146, 120, 236, 77, 165, 0))];
+  TypeMap mtm = un_tsi.getMinimalTypeMap();
+  TypeObject c_to_m_mto;
+  if (!tls->complete_to_minimal_typeobject(cto, c_to_m_mto)) {
+    ASSERT_EQ(1,0);
+  }
+  TypeObject mto2 = mtm[TypeIdentifier(EK_MINIMAL, EquivalenceHashWrapper(197, 93, 228, 89, 233, 88, 151, 207, 208, 98, 147, 32, 200, 147))];
+  ASSERT_EQ(mto2, c_to_m_mto);
+}
+
+int main(int argc, char* argv[])
+{
+  ::testing::InitGoogleTest(&argc, argv);
+    ::DDS::DomainParticipantFactory_var dpf = TheParticipantFactoryWithArgs(argc, argv);
+  dp = dpf->create_participant(184,
+                               PARTICIPANT_QOS_DEFAULT,
+                               0,
+                               0);
+  OpenDDS::DCPS::DomainParticipantImpl* dp_impl =
+    dynamic_cast<OpenDDS::DCPS::DomainParticipantImpl*>(dp.in());
+  tls = dp_impl->get_type_lookup_service();
```

Comment:
I don't think this can be a unit test if it's creating a participant, especially if the test cases are sharing that participant.

Reply:
I'm pretty sure this should go in with compiler tests.